core/src/main/java/org/apache/iceberg/BaseTransaction.java

```diff
@@ -368,14 +368,7 @@ private void commitSimpleTransaction() {
               2.0 /* exponential */)
           .onlyRetryOn(CommitFailedException.class)
           .run(underlyingOps -> {
-            if (base != underlyingOps.refresh()) {
-              this.base = underlyingOps.current(); // just refreshed
-              this.current = base;
-              for (PendingUpdate update : updates) {
```

Comment:
Initially thought of moving the update outside the if check, by direct write test case (`testTransactionRewriteManifestsAppendedDirectly`) will fail in that case.

Hence, falling back to the base metadata when the commit is failed. So, that updates will be added again.

Reply:
Yeah, reapplying updates needs to happen after refreshing the underlying metadata on each attempt.